apps/api/src/routes/v1/dashboardData.ts

```diff
@@ -144,22 +144,65 @@ const route = createRoute({
 });
 
 export default new OpenAPIHono().openapi(route, async c => {
-  const chainStatsQuery = await getChainStats();
-  const dashboardData = await getDashboardData();
-  const networkCapacity = await getNetworkCapacity();
-  const networkCapacityStats = await getProviderGraphData("count");
-  const latestBlocks = await getBlocks(5);
-  const latestTransactions = await getTransactions(5);
 
-  const chainStats = {
-    height: latestBlocks[0].height,
-    transactionCount: latestBlocks[0].totalTransactionCount,
-    ...chainStatsQuery
-  };
+  let chainStats;
+  let now;
+  let compare;
+  let networkCapacity;
+  let networkCapacityStats;
+  let latestBlocks;
+  let latestTransactions;
+
+  try {
+    const chainStatsQuery = await getChainStats();
+    chainStats = { ...chainStatsQuery };
+  } catch (error) {
+    chainStats = {}
+    console.error("Failed to fetch getChainStats:", error);
+  }
+
+  try {
```

Comment:
what would you say if we added smth like this:


```typescript
async function runOrLog<T>(cb: () => Promise<T>, defaultValue: T): Promise<T> {
// or async function runOrLog<T>(cb: () => Promise<T>, defaultValue: T): Promise<T> {
  try {
    return await cb();
  } catch (error) {
    console.error("Error occurred:", error);
    // or report to sentry

    return defaultValue;
  }
}

// to later be used like
const { now, compare } = await runOrLog(getDashboardData, {})
const networkCapacityStats = await runOrLog(getProviderGraphData("count"), 0);
```

it would clean the code up quite a bit